### PR TITLE
Changeform variations

### PR DESCRIPTION
**Pull request by farridav**
Adds in 4 new variations of the change form templates, controlled via `changeform_format` and `changeform_format_overrides` settings

All variants will take you to the correct section in case of an error, all variants use the page hash to navigate back to that section on page refresh

Aims to add new functionality and jazzyness

@RamezIssac please check out this branch and let me know if the `single` option is what your looking for

- single
Render the form out in one page, including inlines, plain and simple, closest to the original Django admin change form

changeform_single

- horizontal_tabs
Puts all fieldsets and inlines into tab panes with horizontal nav tab controls, this is the default view for change 
forms that have fieldsets. or an inline

changeform_horizontal_tabs

- vertical_tabs
Puts each fieldset or inline in a separate pane, controlled by vertical tabs on the left hand side.

changeform_vertical_tabs

- collapsible
Puts all fieldsets and inlines in bootstrap collapsables in an accordion, allows many collapsables to be open at the 
same time, the first collapsible is opened

changeform_accordion

- carousel
Puts fieldsets and inlines into a bootstrap carousel, and allows paginaton with previous/nect buttons, as well as an indicators.

changeform_carousel

Future enhancements:
- Allow vertical tabs to be on the left or right
- More change form templates !!

Fixes https://github.com/farridav/django-jazzmin/issues/84
Fixes https://github.com/farridav/django-jazzmin/issues/86
Fixes https://github.com/farridav/django-jazzmin/issues/83

### CODE REVIEW

**Comment by farridav** [comment]
Fixes #86 #84 #83

**Comment by RamezIssac** [comment]
Bravo !!!! :clap: 
This is more then what i was proposing at first.. the screens looks amazing and the options you provided ... Nice :+1:

**Comment by farridav** [comment]
@TysonRV rebased

File: jazzmin/static/jazzmin/js/change_form.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    // Style the inline fieldset button
+    $('.inline-related fieldset.module .add-row a').addClass('btn btn-sm btn-default float-right');
+
+    // Ensure we preserve the tab the user was on using the url hash, even on page reload
+    if ($tabs.length) {handleTabs($tabs);}
+    else if ($carousel.length) {handleCarousel($carousel);}
+    else if ($collapsible.length) {handleCollapsible($collapsible);}
+});
+
```
Comment (TysonRV) [inline jazzmin/static/jazzmin/js/change_form.js:97]: This works great!

File: jazzmin/templates/admin_doc/view_index.html [hunk 2]
```diff
@@ -18,13 +18,13 @@
...
 
-<div id="accordion">
+<div id="jazzy-collapsible">
 
   {% for ns_views in views_by_ns %}
     <div class="card card-primary">
       <div class="card-header">
         <h3 id="ns|{{ ns_views.grouper }}" class="card-title">
-            <a data-toggle="collapse" data-parent="#accordion" href="#collapse-{{ ns_views.grouper }}" class="" aria-expanded="true">
+            <a data-toggle="collapse" data-parent="#collapsible" href="#collapse-{{ ns_views.grouper }}" class="" aria-expanded="true">
```
[the remark this answers is not in the document]
Comment (TysonRV) [reply jazzmin/templates/admin_doc/view_index.html:27]: Oh, I see, it's just changing that by `jazzy-collapsible`

**Comment by TysonRV** [comment]
:shipit:

**Pull request merged**